test/acceptance/features/environment.py

```diff
@@ -49,9 +55,28 @@ def before_all(_context):
         _context.sbo_namespace = output
     else:
         assert False, f"TEST_ACCEPTANCE_START_SBO={start_sbo} is currently unsupported."
+    _context.no_scenarios_failed = True
 
 
 def before_scenario(_context, _scenario):
     _context.bindings = dict()
     output, code = cmd.run(f'{ctx.cli} get ns default -o jsonpath="{{.metadata.name}}"')
     assert code == 0, f"Checking connection to OS cluster by getting the 'default' project failed: {output}"
+
+
+def after_scenario(_context, scenario):
+    namespace = os.getenv("TEST_NAMESPACE")
+    if delete_namespace == "alwaysdelete":
+        output, code = cmd.run(f"{ctx.cli} delete namespace {namespace}.")
+        assert code == 0, f"Deletion of namespace failed: {output}"
+        print(f"Namespace {namespace} deleted.")
+    elif (delete_namespace == "keepwhenfailed"):
+        if scenario.status == Status.passed and (_context.no_scenarios_failed):
+            output, code = cmd.run(f"{ctx.cli} delete namespace {namespace}.")
+            assert code == 0, f"Deletion of namespace failed: {output}"
+            print(f"Namespace {namespace} deleted.")
```

Comment:
Please do not repeat the code for deleting the namespace. You can extract it to a separate function and then call it instead, for instance.

Reply:
Additionally, if by any chance the namespace is not there, it should not fail the test. After all, the point of this is to ensure that the namespace is not there after each scenario.

```suggestion
            output, code = cmd.run(f"{ctx.cli} delete namespace {namespace} --ignore-not-found")
            assert code == 0, f"Deletion of namespace failed: {output}"
            print(f"Namespace {namespace} deleted.")
```